js/framework/AnalyticsConstants.js

```diff
@@ -9,37 +9,356 @@ if (!OO.Analytics.EVENTS)
+     */
     VIDEO_PLAYING:                  'video_playing',
+
+    /**
+     * @event Analytics#EVENTS#VIDEO_PAUSED
+     * @description This message is sent when video playback has paused.
+     */
     VIDEO_PAUSED:                   'video_paused',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_ENDED
+     * @description This message is sent when video playback has completed. This
+     * includes finishing playback of all ads.
+     */
     VIDEO_ENDED:                    'video_ended',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_REPLAY_REQUESTED
+     * @description This message is sent after VIDEO_ENDED, when the player is
+     * requested to start video playback from the beginning of the video. This
+     * is only the request, not when the video actually start playing again.
+     */
     VIDEO_REPLAY_REQUESTED:         'video_replay_requested',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_SOURCE_CHANGED
+     * @description This message when the player has received a new video source
+     * to load.  This will happen when the first video initially starts to load,
+     * when switching video sources during playback and also when switching to a
+     * new video after VIDEO_ENDED.  This will not be received on VIDEO_REPLAY_REQUESTED.
+     * @param {Array} paramArray Array of length 1, containing an instance of
+     * OO.Analytics.EVENT_DATA.VideoSourceData
+     */
     VIDEO_SOURCE_CHANGED:           'video_source_changed',
+
+    /**
+     * @event Analytics#EVENTS#VIDEO_STREAM_METADATA_UPDATED
+     * @description This message is sent when video stream metadata has been
+     * downloaded.  In contains information about the stream and metadata
+     * for any plugins that should be loaded.
+     * @param {Array} paramArray Array of length 1, contains an object holding all
+     * the metadata for each plugin that should be loaded.
+     */
     VIDEO_STREAM_METADATA_UPDATED:  'video_stream_metadata_updated',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_CONTENT_METADATA_UPDATED
+     * @description This message is sent when the video content data has been
+     * downloaded. This will contain information about the video content. For
+     * example, title and description.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoContentMetadata.
+     */
     VIDEO_CONTENT_METADATA_UPDATED: 'video_content_metadata_updated',
-    VIDEO_SEEK_STARTED:             'video_seek_requested',
-    VIDEO_SEEK_ENDED:               'video_seek_completed',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_SEEK_STARTED
+     * @description This message is sent when a video seek is requested.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoSeekStartedData.
+     */
+    VIDEO_SEEK_STARTED:             'video_seek_started',
```

Comment:
what if the seek is requested but never happens?   why change it from requested to started?  You could attempt to seek an unseekable video with APIs and it just wouldn't do anything